test/e2e/e2e_suite_test.go

```diff
@@ -208,6 +214,38 @@ func initBootstrapCluster(bootstrapClusterProxy framework.ClusterProxy, config *
 	}, config.GetIntervals(bootstrapClusterProxy.GetName(), "wait-controllers")...)
 }
 
+func dumpBootstrapClusterLogs(bootstrapClusterProxy framework.ClusterProxy) {
+	if bootstrapClusterProxy == nil {
+		return
+	}
+
+	clusterLogCollector := bootstrapClusterProxy.GetClusterLogCollector()
+	if clusterLogCollector == nil {
+		return
+	}
+
+	nodes, err := bootstrapClusterProxy.GetClientSet().CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```

Comment:
let's wait for https://github.com/kubernetes-sigs/cluster-api/pull/4706 and use the global ctx object

Reply:
https://github.com/kubernetes-sigs/cluster-api/pull/4706 just merged so it now possible to address this comment

Reply:
Done!